tests/unit/test-firefox/test.remote.js

```diff
@@ -375,4 +376,14 @@ describe('firefox.remote', () => {
 
   });
 
+  describe('defaultRemotePortFinder', () => {
+
+    it('resolves to an open port', () => {
```

Comment:
We could improve this test a little, e.g. first simple improvement is to rewrite it as an [async function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function).

Besides the small stylistic tweak suggested above, there is another detail that we could improve related to the "coverage" of the test (not in terms of lines covered by the test, but in terms of how much it is able to catch regression in the implementation):
we could add an additional assertion that double-checks that the port number we got works is going to work for a user.
e.g. a [user can't allocate ports in the range between 0 and 1023](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?&page=2) and so it may be reasonable to require that `findFreeTcpPort` should return a port number >= 1024
(which should also be true for our current implementation).

----

Some additional details about why we may want to add the assertion above (and maybe even increase the test, even as a follow up)

Right now `findFreeTcpPort` is just a small wrapper around a feature implemented by the nodejs native `net` module, and we trust that nodejs native `net` module is returning what we expect (a free tcp port) and it is tested on its own.

But let's make the hypothesis that at some point `findFreeTcpPort` is rewritten to not use the native `net` module, let's say that it is rewritten as:
```
export function findFreeTcpPort() { return 0; }
```
(which is clearly unrealistic, but it is just an example of the idea).

The current version of this test would not fail in that case, but it would fail with the additional assertion suggested above.

Additional test coverage could be introduced by:

- create a tcp server listening on the tcp port we got from the first `findFreeTcpPort` call (which double-check that the port was actually free and an application running by a user can listen on it)
- call `findFreeTcpPort` again and check that we got a different port, which is still >= 104.

Reply:
This is a great explanation! Thank you so much for this!